electrum_nmc/electrum/gui/qt/main_window.py

```diff
@@ -3348,10 +3348,15 @@ def create_buy_names_tab(self):
         self.buy_names_ui = Ui_BuyNamesPage()
         self.buy_names_ui.setupUi(w)
 
-        # TODO: allow hex names
-        self.buy_names_new_name_lineedit = self.buy_names_ui.registerName
-        self.buy_names_new_name_lineedit.textChanged.connect(self.update_buy_names_preview)
+        self.buy_names_new_name_ascii_lineedit = self.buy_names_ui.registerNameAscii
+        self.buy_names_new_name_ascii_lineedit.textChanged.connect(self.update_register_name_from_ascii)
+        
+        self.buy_names_new_name_hex_lineedit = self.buy_names_ui.registerNameHex
+        self.buy_names_new_name_hex_lineedit.textChanged.connect(self.update_buy_names_preview)
 
+        self.buy_names_new_name_domain_lineedit = self.buy_names_ui.registerNameDomain
+        self.buy_names_new_name_domain_lineedit.textChanged.connect(self.update_register_name_and_domain)
```

Comment:
I think we want `textEdited` rather than `textChanged` to avoid loops. See the Qt5 documentation for details on the difference.